### ISSUE TITLE
QRCode crashes when taking picture

### ISSUE THREAD

**Issue by SnelleJelle** [Truiton/MobileVisionAPI#3]
I can start the app just fine. When i press "take a picture" it crashes.
I have not made any changes, I was just trying it out to see what it did.

Stacktrace:

```E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: com.truiton.mobile.vision.qrcode, PID: 3685
                  java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=20, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {com.truiton.mobile.vision.qrcode/com.truiton.mobile.vision.qrcode.MainActivity}: android.os.FileUriExposedException: file:///storage/emulated/0/picture.jpg exposed beyond app through ClipData.Item.getUri()
                      at android.app.ActivityThread.deliverResults(ActivityThread.java:4089)
                      at android.app.ActivityThread.handleSendResult(ActivityThread.java:4132)
                      at android.app.ActivityThread.-wrap20(ActivityThread.java)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533)
                      at android.os.Handler.dispatchMessage(Handler.java:102)
                      at android.os.Looper.loop(Looper.java:154)
                      at android.app.ActivityThread.main(ActivityThread.java:6119)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
                   Caused by: android.os.FileUriExposedException: file:///storage/emulated/0/picture.jpg exposed beyond app through ClipData.Item.getUri()
                      at android.os.StrictMode.onFileUriExposed(StrictMode.java:1799)
                      at android.net.Uri.checkFileUriExposed(Uri.java:2346)
                      at android.content.ClipData.prepareToLeaveProcess(ClipData.java:845)
                      at android.content.Intent.prepareToLeaveProcess(Intent.java:8941)
                      at android.content.Intent.prepareToLeaveProcess(Intent.java:8926)
                      at android.app.Instrumentation.execStartActivity(Instrumentation.java:1517)
                      at android.app.Activity.startActivityForResult(Activity.java:4225)
                      at android.support.v4.app.BaseFragmentActivityJB.startActivityForResult(BaseFragmentActivityJB.java:48)
                      at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:75)
                      at android.app.Activity.startActivityForResult(Activity.java:4183)
                      at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:856)
                      at com.truiton.mobile.vision.qrcode.MainActivity.takePicture(MainActivity.java:156)
                      at com.truiton.mobile.vision.qrcode.MainActivity.onRequestPermissionsResult(MainActivity.java:73)
                      at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:7084)
                      at android.app.Activity.dispatchActivityResult(Activity.java:6936)
                      at android.app.ActivityThread.deliverResults(ActivityThread.java:4085)
                      at android.app.ActivityThread.handleSendResult(ActivityThread.java:4132) 
                      at android.app.ActivityThread.-wrap20(ActivityThread.java) 
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1533) 
                      at android.os.Handler.dispatchMessage(Handler.java:102) 
                      at android.os.Looper.loop(Looper.java:154) 
                      at android.app.ActivityThread.main(ActivityThread.java:6119) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) ```

**Comment by ZKjellberg** [comment]
I believe this is related to the issue with [file:// scheme is now not allowed to be attached with Intent on targetSdkVersion 24.](https://inthecheesefactory.com/blog/how-to-share-access-to-file-with-fileprovider-on-android-nougat/en)

**Comment by githubtoanony** [comment]
add the StrictMode vmpolicy in the activity as
```
 StrictMode.VmPolicy.Builder builder = new StrictMode.VmPolicy.Builder();
 StrictMode.setVmPolicy(builder.build());
```

**Comment by mohitgupt** [comment]
Resolved with commit 70cf31841ca68ea29d2b1ddad17a64398b03bd11.
